### PR TITLE
DEPR: deprecate Index.is_floating

### PR DESCRIPTION
**Pull request by yuuun**
- [ ] progress towards https://github.com/pandas-dev/pandas/issues/50042
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/v2.0.0.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Comment by yuuun** [comment]
@topper-123  Hi, Can you check if I have done right?

**Review by topper-123 — APPROVED** [review]
LGTM. I’ll metge when the other ralated PRs get pulled in, so this may sit for a while

**Comment by yuuun** [comment]
@topper-123  I just clicked "update branch" and some failing tasks have appeared.. Do you have any solution for this? It's little bit hard for me to fix this issue..

**Review by phofl — APPROVED** [review]

**Comment by phofl** [comment]
thx @yuuun

File: pandas/core/indexes/base.py [hunk 1]
```diff
@@ -2297,6 +2300,12 @@ def is_floating(self) -> bool:
         >>> idx.is_floating()
         False
         """
+        warnings.warn(
+            f"{type(self).__name__}.is_floating is deprecated."
+            "Use pandas.api.types.is_float_dtype instead",
```
Comment (simonjayhawkins) [inline pandas/core/indexes/base.py:2305]: probably could do with a space here.

**Pull request merged**